eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -100,20 +103,152 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
+    pipeline = get_filter_group_count_pipeline(taxonomy, assembly_list)
+    cve_res = get_rs_count_per_assembly_in_collection(mongo_source, 'clusteredVariantEntity', pipeline)
+    dbsnp_res = get_rs_count_per_assembly_in_collection(mongo_source, 'dbsnpClusteredVariantEntity', pipeline)
+
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        for assembly in assembly_list:
+            sources, rs_count = get_sources_and_rs_count(cve_res, dbsnp_res, assembly)
+            entry_exists = check_if_entry_exists_for_taxonomy_and_assembly(pg_conn, taxonomy, assembly)
+            if entry_exists:
+                update_rs_count_for_taxonomy_assembly(pg_conn, rs_count, taxonomy, assembly)
+            else:
+                insert_new_entry_for_taxonomy_assembly(pg_conn, sources, rs_count, release_version, taxonomy, assembly,
+                                                       reference_directory)
+
+
+def get_assembly_list_for_taxonomy(private_config_xml_file, taxonomy):
+    assembly_list = []
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        query = f'SELECT assembly_accession from evapro.assembly where taxonomy_id = {taxonomy}'
+        for assembly in get_all_results_for_query(pg_conn, query):
+            assembly_list.append(assembly[0])
+    return assembly_list
+
+
+def get_filter_group_count_pipeline(taxonomy, assembly_list):
+    filter_by_taxonomy_and_assembly = {
+        "$match":
+            {
+                "tax": {
+                    "$eq": taxonomy,
+                },
+                "asm": {
+                    "$in": assembly_list,
+                }
+            }
+    }
+    group_by_assembly = {
+        "$group":
+            {
+                "_id": {"assembly": "$asm"},
+                "rs_count": {"$sum": 1},
+            }
+    }
+    return [filter_by_taxonomy_and_assembly, group_by_assembly]
+
+
+def get_rs_count_per_assembly_in_collection(mongo_source, coll_name, pipeline):
+    coll = mongo_source.mongo_handle[mongo_source.db_name][coll_name]
+    agg_result = coll.aggregate(pipeline, batchSize=0)
+    agg_res_dict = {}
+    for entry in agg_result:
+        agg_res_dict[entry['_id']['assembly']] = entry['rs_count']
+    return agg_res_dict
+
+
+def get_sources_and_rs_count(cve_res, dbsnp_res, assembly):
+    if assembly in cve_res and assembly in dbsnp_res:
+        return ['DBSNP, EVA', cve_res[assembly] + dbsnp_res[assembly]]
+    elif assembly in cve_res and assembly not in dbsnp_res:
+        return ['EVA', cve_res[assembly]]
+    elif assembly not in cve_res and assembly in dbsnp_res:
+        return ['DBSNP', dbsnp_res[assembly]]
+    else:
+        # if there is an assembly in evapro but there is no clustered variant in any of the collections,
+        # an entry will still be created in clustering tracking table with sources as None and rs count 0
+        return ['None', 0]
```

Comment:
For this case I would avoid creating a row at all.

Reply:
done